File: kitchen/kitchen/custom/python/delivery_note.py
import frappe
from frappe.model.mapper import get_mapped_doc
@frappe.whitelist()
def make_packing_slip(source_name, target_doc=None):
	doclist = get_mapped_doc(
		"Delivery Note",
		source_name,
		{
			"Delivery Note": {
				"doctype": "Packing Slip",
				"field_map": {"name": "delivery_note", "letter_head": "letter_head"},
				"validation": {"docstatus": ["=", 1]},
			}
		},
		target_doc,
	)

	return doclist

File: kitchen/kitchen/custom/js/delivery_note.js
frappe.ui.form.on('Delivery Note', {
    refresh: function(frm,cdt,cdn) {
        if(frm.doc.docstatus == 1){ 
        frm.add_custom_button(__('Packing Slip'), function(){
            frappe.model.open_mapped_doc({
                method: "kitchen.kitchen.custom.python.delivery_note.make_packing_slip",
                frm:frm,
        })
    }, __("Create"));
  }
  if(frm.doc.docstatus == 0){
    frm.add_custom_button(__('Packing Slip'), function(){
        frappe.msgprint("Please submit the document to proceed further")
    },__("Create"));
}
}});